### ISSUE TITLE
`PossiblyUnusedMethod` false positive for controller methods registered via array-callable `[Controller::class, "method"]` in route definitions

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#1071]
## Describe the bug

Controller methods referenced via the standard array-callable syntax produce `PossiblyUnusedMethod` false positives even though `findUnusedCode=true` and `routes/` is included in `projectFiles`.

```
ERROR: PossiblyUnusedMethod
Cannot find any calls to method App\Http\Controllers\FooController::index (see https://psalm.dev/087)
    public function index(Request $request): View
```

## Code to reproduce

**`routes/web.php`**

```php
use App\Http\Controllers\FooController;
use Illuminate\Support\Facades\Route;

Route::get('foo', [FooController::class, 'index'])->name('foo.index');
Route::post('foo', [FooController::class, 'store'])->name('foo.store');
```

**`app/Http/Controllers/FooController.php`**

```php
<?php declare(strict_types=1);

namespace App\Http\Controllers;

use Illuminate\Contracts\View\View;
use Illuminate\Http\Request;

final readonly class FooController
{
    public function index(Request $request): View
    {
        return view('foo.index');
    }

    public function store(Request $request): \Illuminate\Http\RedirectResponse
    {
        return redirect()->route('foo.index');
    }
}
```

Both `index` and `store` are reported as `PossiblyUnusedMethod`.

## Root cause analysis

`Illuminate\Routing\Router::get()` (and `post`, `put`, `patch`, `delete`) is typed:

```php
/**
 * @param  array|string|callable|null  $action
 */
public function get($uri, $action = null)
```

`array` appears before `callable` in the union. When Psalm sees `[FooController::class, 'index']` passed here, it matches the `array` branch rather than the `callable` branch, so it does not record a method reference to `FooController::index`. The method therefore appears unused to Psalm's used-symbol pass.

Issue #876 notes that the `[Controller::class, 'method']` array-callable form is "handled by Psalm natively". In practice it is not — at least not for `Route::get/post/...` where the parameter is typed as `array|string|callable|null`.

The fix is for the plugin to emit a synthetic method reference whenever it sees `Route::<verb>($uri, [SomeClass::class, 'someMethod'])` (where the second element is a string literal), analogously to what #876 proposes for `Route::controller()` and `Route::resource()`.

## Impacted Versions

```text
laravel/framework       v12.62.0
psalm/plugin-laravel    dev-master
vimeo/psalm             7.0.0-beta19
```

## Psalm config

```xml
<psalm
    findUnusedCode="true"
    findUnusedVariablesAndParams="true"
    errorLevel="1"
>
    <plugins>
        <pluginClass class="Psalm\LaravelPlugin\Plugin"/>
    </plugins>
    <projectFiles>
        <directory name="app"/>
        <directory name="routes"/>
        <ignoreFiles>
            <directory name="vendor"/>
        </ignoreFiles>
    </projectFiles>
</psalm>
```

## Additional context

Related: #876 (covers `Route::controller()` / `Route::resource()` string-action forms — the sibling gap for array-callable forms).

**Comment by alies-dev** [comment]
Closing as duplicate of #876 — the array-callable form is part of the same gap in route-action reference tracking.
